src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -1565,6 +1580,12 @@ deploy_transaction_execute (RpmostreedTransaction *transaction, GCancellable *ca
     return FALSE;
   changed = changed || layering_changed;
 
+  /* When using containers and nothing changed after prep_layering, return early.
+   * This happens when all requested packages are already present in the container image.
+   * For idempotent mode, this avoids the "No packages in transaction" error. */
+  if (skip_base_check && !layering_changed && !self->refspec)
```

Comment:
This check is causing others tests to regress like layering-local, changing !layering_changed for !changed makes the layering-local pass but breaks the net idempotent-layering test.